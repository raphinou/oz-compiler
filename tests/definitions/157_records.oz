% Records on LHS of unification, both const and not const
local
   R A B C L FA FB
   R2 R3
in
   FA=a
   A=1
   FB=b
   B=2
   L=lab
   L(FA:A FB:B )=R

   {Show R}

   C=3
   rec(a FB:b c:C)=R2
   {Show R2}

   rec(a:1 b:2)=R3
   {Show R3}
end
