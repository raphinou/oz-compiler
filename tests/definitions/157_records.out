lab(a:1 b:2)
rec(1:a b:b c:3)
rec(a:1 b:2)
